### PR TITLE
Added speech tests

### PR DESCRIPTION
**Pull request by EvanBacon**
# How

- Fixed speech event
- Added tests for speech
- Added API for `ExponentTest`
- Added `getVoices` to `expo-speech`

# Test Plan

- CI should pass with new tests

### CODE REVIEW

**Review by ide — APPROVED** [review]

File: apps/test-suite/puppeteer-test/index.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ const manuallyRunWebpack = true;
 
 let server;
 
+if (typeof process.env.CI === 'undefined') {
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply apps/test-suite/puppeteer-test/index.js]: If the env isn't defined then it uses `process.stdout.isTTY`. This seems like the correct logic. If it's defined then we do not want to use a fallback value.

[the remark this answers is not in the document]
Comment (ide) [reply apps/test-suite/puppeteer-test/index.js]: If someone runs the tests with `CI=0` that seems to indicate that they don't want to run it in CI mode.

File: apps/test-suite/tests/Localization.js [hunk 2]
```diff
@@ -84,14 +85,16 @@ export function test(t) {
...
-    t.it('Gets the current timzezone', async () => {
-      const result = Localization.timezone;
-      t.expect(result).toBeDefined();
-      t.expect(typeof result).toBe('string');
-      t.expect(result.length > 0).toBe(true);
-      // Format: expect something like America/Los_Angeles or America/Chihuahua
-      t.expect(result.split('/').length > 1).toBe(true);
-    });
+    if (Platform.OS !== 'web') {
+      t.it('Gets the current timzezone', async () => {
```
Comment (ide) [inline apps/test-suite/tests/Localization.js]: fix this while you're editing this anyway

```suggestion
      t.it('Gets the current timezone', async () => {
```

File: packages/expo-speech/src/Speech/Speech.types.ts [hunk 3]
```diff
@@ -29,3 +29,10 @@ export type Voice = {
   quality: VoiceQuality;
   language: string;
 };
+
+export type WebVoice = Voice & {
+  isDefault: boolean; 
+  localService: boolean; 
+  name: string; 
+  voiceURI: string;
+}
```
Comment (ide) [inline packages/expo-speech/src/Speech/Speech.types.ts]:
```suggestion
};

```

File: packages/expo-speech/src/Speech/Speech.types.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export type SpeechOptions = {
   onStart?: () => void | SpeechEventCallback;
   onStopped?: () => void | SpeechEventCallback;
   onDone?: () => void | SpeechEventCallback;
-  onError?: (error: Error) => void | SpeechEventCallback;
+  onError?: (error: SpeechSynthesisErrorEvent) => void | SpeechEventCallback;
```
Comment (ide) [inline packages/expo-speech/src/Speech/Speech.types.ts]: this is not a SpeechSynthesisErrorEvent on Android and iOS

**Review by ide — APPROVED** [review]

**Review by ide — APPROVED** [review]

**Pull request merged**